### DISCUSSION TITLE
SOLVED Smart Connect - Notes_ customGPT points to Brians vault (even though I use my key)

### DISCUSSION THREAD

**Discussion by ottantaventi** [brianpetro/smart-connect#59] [Help & Support] [answered]
I spent the whole day trying to figure out to make Smart Connect, Smart Connection and ChatGPT work and I succeded only partially.

 **The Smart Connect ChatGPT does NOT connect to my vault, but to one of Brians vault**
 Before you ask: yes I'm using a purchased key.
 Here I used a clean new vault only 6 notes in it and when I ask the bot _What's my next action for a project?_ (one of the default queries that are proposed when I open the custom GPT, I get the following answer which clearly points to one of Brian's vaults:
 
![image](https://github.com/user-attachments/assets/e16c34de-d29a-4b88-9953-fab51e9ddae7)

To further convince you, I asked to create a note in your vault:
![image](https://github.com/user-attachments/assets/aa7fd497-c956-42c9-a2fe-bfb0802599de)

You should see it in your inbox :-)
https://connect.smartconnections.app/open?vault=obsidian-1&file=%2B📥%20inbox%2FHi%20Brian%20-%20my%20name%20is%20Alex.md

Out of desperation I tried to install a customGPT:

![image](https://github.com/user-attachments/assets/d1b12f69-018d-43b8-ae44-5183089eedfa)

and today it seems to work a little bit better than this morning and yesterday (I guess your actually working on it right now) and I've been able to:
- summarize 2 notes
- create a new note out of it
- make the bot update a section of the note
This to say that I was able to connect somehow Smart Connect, to ChatGPT and to my Obsidian vault.

What frustrated me yesterday and this morning is that when I ask about how many notes are present in the system (this happened both with my real vault, then also with the test vault containing only 6 notes) the bot answers they are less than in reality.
In the test vault I had initially only 1 note, than I adde 5 more: sometimes the bot answered there where all 6 (i even asked him to list them) and sometimes he went back saying there were only 1 (and when i asked to list all the notes in the system he listed only the first note, the welcome note).

**Comment by brianpetro** [comment, marked as answer]
Thanks for bringing this to my attention and taking the time to clarify the issue.

Security is an important part of Smart Connect, so I looked into this immediately. 

Below is a brief report with my findings

![Screenshot 2024-12-03 at 3 58 57 PM](https://github.com/user-attachments/assets/76cb732d-00d4-4608-81bd-52124d87ecb5)

Update: @ottantaventi followed up via email and confirmed the above hypothesis. Marking as SOLVED. 
🌴
